ee/http/middleware/pat.go

```diff
@@ -2,26 +2,69 @@ package middleware
 
 import (
 	"net/http"
+	"time"
 
+	"github.com/uptrace/bun"
+	"go.signoz.io/signoz/pkg/types"
 	"go.signoz.io/signoz/pkg/types/authtypes"
+	"go.uber.org/zap"
 )
 
 type Pat struct {
+	db      *bun.DB
 	uuid    *authtypes.UUID
 	headers []string
 }
 
-func NewPat(headers []string) *Pat {
-	return &Pat{uuid: authtypes.NewUUID(), headers: headers}
+func NewPat(db *bun.DB, headers []string) *Pat {
+	return &Pat{db: db, uuid: authtypes.NewUUID(), headers: headers}
 }
 
 func (p *Pat) Wrap(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		var values []string
+		var patToken string
+		var pat types.StorablePersonalAccessToken
+		var updateLastUsed bool
+
 		for _, header := range p.headers {
 			values = append(values, r.Header.Get(header))
```

Comment:
this thing I am a bit confused whether to even keep these headers in context now